File: 0724-find-pivot-index/0724-find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        
        int totalSum = 0;
        for(int i = 0; i < nums.size(); i++){
             totalSum += nums[i];
        }

        int leftSum = 0;
        int rightSum = 0;
        int ans = 0;
        for(int j = 0; j < nums.size(); j++){
               rightSum = totalSum - leftSum - nums[j];

               if(leftSum == rightSum){
                   return j;
               } else{
                   leftSum += nums[j];
               }
        }
        return -1;
    }
};

File: 0724-find-pivot-index/0724-find-pivot-index.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var pivotIndex = function(nums) {
     var totalSum = 0;
        for(var i = 0; i < nums.length; i++){
             totalSum += nums[i];
        }

        var leftSum = 0;
        var rightSum = 0;
        var ans = 0;
        for(var j = 0; j < nums.length; j++){
               rightSum = totalSum - leftSum - nums[j];

               if(leftSum == rightSum){
                   return j;
               } else{
                   leftSum += nums[j];
               }
        }
        return -1;
};